cmd/subnetcmd/upgradecmd/install.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package upgradecmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/ava-labs/avalanche-cli/pkg/binutils"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanche-cli/pkg/subnet/upgrades"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+	ANRclient "github.com/ava-labs/avalanche-network-runner/client"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+)
+
+const (
+	timestampFormat  = "20060102150405"
+	tmpSnapshotInfix = "-tmp-"
+)
+
+// avalanche subnet upgrade generate
+func newUpgradeInstallCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "install [subnetName]",
+		Short: "Installs upgrade bytes onto subnet nodes",
+		Long:  `Installs upgrade bytes onto VMs.`, // TODO fix this wording
+		RunE:  installCmd,
+		Args:  cobra.ExactArgs(1),
+	}
+	return cmd
+}
+
+func installCmd(cmd *cobra.Command, args []string) error {
+	subnetName := args[0]
+
+	if !app.SubnetConfigExists(subnetName) {
+		return errors.New("subnet does not exist")
+	}
+
+	sc, err := app.LoadSidecar(subnetName)
+	if err != nil {
+		return fmt.Errorf("unable to load sidecar: %w", err)
+	}
+
+	networkToUpgrade, err := selectNetworkToUpgrade(sc)
+	if err != nil {
+		return err
+	}
+
+	switch networkToUpgrade {
+	// in this case, we just are going to generate new update bytes
+	case futureDeployment:
+		return upgradeGenerateCmd(cmd, args)
+	// update a locally running network
+	case localDeployment:
+		return localInstallNetworkUpgrades(subnetName, sc)
+	}
+
+	return nil
+}
+
+func localInstallNetworkUpgrades(subnetName string, sc models.Sidecar) error {
+	// For a already deployed subnet, the supported scheme is to
+	// save a snapshot, and to load the snapshot with the upgrade
```

Comment:
Can we not just restart the nodes? Why do we need to mess with snapshots?

Reply:
Because we need to pass a new flag pointing to the update bytes file to ANR. ANR before this doesn't know anything about the existence of this file.

Currently the only supported way is to restart from a snapshot and provide the new flags via `LoadSnapshot`.

Alternatively, we could add a new endpoint to ANR which would do this behind the scenes somehow, but that would require a new ANR release. cc @felipemadero